models/cartDao.js

```diff
@@ -0,0 +1,93 @@
+const appDataSource = require('./dataSource');
+
+const createCart = async (userId, productId, sizeId) => {
+  const insertCart = await appDataSource.query(`
+    INSERT INTO carts 
+    (
+      user_id,
+      product_option_id
+    ) 
+    SELECT ${userId}, (SELECT po.id FROM product_options AS po
+    JOIN products AS p ON po.product_id = p.id
+    JOIN sizes AS s ON po.size_id = s.id
+    WHERE po.product_id=${productId} AND po.size_id=${sizeId}) 
+    WHERE NOT EXISTS
+    (
+      SELECT * FROM carts AS c 
+      WHERE c.user_id=${userId} AND c.product_option_id= 
+        (
+          SELECT po.id FROM product_options AS po
+          JOIN products AS p ON po.product_id = p.id
+          JOIN sizes AS s ON po.size_id = s.id
+          WHERE po.product_id=${productId} AND po.size_id=${sizeId}
+        )  
+    )
+  `)
+  
+  if (insertCart.affectedRows === 0) {
+    const error = new Error('FAILED');
+    error.statusCode = 400;
+    throw error;
+  }
+  return insertCart;
+}
+
+const getCart = async (userId) => {
+  const result = await appDataSource.query(`
+    SELECT
+      c.user_id AS userId,
+      c.product_option_id AS productOptionId,
+      c.count,
+      po.product_id AS productId,
+      po.size_id AS sizeId,
+      p.thumbnail_image_url AS thumbnailUrl,
+      p.name,
+      p.price,
+      s.foot_size AS footSize,
+      po.stock
+    FROM carts AS c
+    JOIN users AS u ON c.user_id = u.id
+    JOIN product_options AS po ON c.product_option_id = po.id
+    JOIN products p ON po.product_id = p.id
+    JOIN sizes AS s ON po.size_id = s.id
+    WHERE c.user_id=?`,
+    [userId]
+  );
+  return result;
+}
```

Comment:
장바구니에 대한 데이터를 보내줄 때 장바구니에 대한 id 값을 포함하여 보내주세요!

데이터베이스 관점에서 보았을 때 데이터를 구분하는 고유의 값은 id로 설정하였습니다.
추가로 cart의 id를 보냈을 때 장바구니를 삭제할 수 있는 로직이 좀 더 간결해질 것 같습니다.


getCart라는 함수 네이밍도 좋지만, 특정 유저의 정보를 통해 장바구니 정보를 얻는다는 의미를 담아 함수 이름을 다음과 같이 수정할 수도 있을 것 같습니다!
```suggestion
const getCartByUserId = async (userId) => {
  const result = await appDataSource.query(`
    SELECT
      c,id,
      c.user_id AS userId,
      c.product_option_id AS productOptionId,
      c.count,
      po.product_id AS productId,
      po.size_id AS sizeId,
      p.thumbnail_image_url AS thumbnailUrl,
      p.name,
      p.price,
      s.foot_size AS footSize,
      po.stock
    FROM carts AS c
    JOIN users AS u ON c.user_id = u.id
    JOIN product_options AS po ON c.product_option_id = po.id
    JOIN products p ON po.product_id = p.id
    JOIN sizes AS s ON po.size_id = s.id
    WHERE c.user_id=?`,
    [userId]
  );
  return result;
}
```